openwisp_controller/config/admin.py

```diff
@@ -52,6 +54,25 @@
     from django.contrib.admin import ModelAdmin
 
 
+def get_system_context_output(obj):
+    if hasattr(obj, 'device'):
+        system_context = obj.device.get_system_context()
+    else:
+        system_context = obj.get_system_context()
+    output = ''
+    for key, value in system_context.items():
+        if '\n' in value:
+            html = f'<p><strong>{key}:</strong> {linebreaks(value)}</p>'
+        else:
+            html = f'<p><strong>{key}:</strong> {value}</p>'
```

Comment:
these lines are mostly the same, eliminate the duplication, also add a css class for the value, eg:

```python
css = "single"
if .....
  value = linebreaks(value)
  css = "multi"
html = f'<p><strong>{key}:</strong><span class="{css}">{value}</p>'
```

Please ensure that multiline value are styled with preformatted style (like the `<pre>` element).